docs/building_the_project.rst

```diff
@@ -448,27 +460,36 @@ To build with the rocRAND backend instead simply replace:
 .. code-block:: bash
 
    -DENABLE_ROCBLAS_BACKEND=True   \
-   -DTARGET_DOMAINS=blas
```

Comment:
@FMarno pointed out that cmake invocation above doesn't specify `TARGET_DOMAINS`. `TARGET_DOMAINS` isn't actually needed, but this PR originally included it for consistency.

In [196dbb9](https://github.com/oneapi-src/oneMKL/pull/457/commits/196dbb924448eb3bc7ec15f78fcfd231922b94a4) I have removed the `TARGET_DOMAIN` references for building roc* with DPC++.